File: chapter6/factorial.c
#include <cs50.h>
#include <stdio.h>

int fact(int num)
{
    if (num == 0)
    {
        return 1;
    }
    else
    {
        return num * fact(num - 1);
    }
}

int main(void)
{
    printf("Please enter a number: ");
    int num = GetInt();
    
    int ans = fact(num);
    
    printf("%d\n", ans);
}

File: chapter6/factorial.py
import cs50

def fact(x):
    if x == 0:
        return 1
    else:
        return x * fact(x-1)
        
def main():
    print("Please enter a number: ", end="")
    num = cs50.get_int()
    
    ans = fact(num)
    print(ans)

if __name__ == "__main__":
    main()